src/enclave/claims.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "crypto/hash_provider.h"
+#include "ds/logger.h"
+
+namespace ccf
+{
+  class ClaimsDigest
+  {
+  public:
+    using Digest = crypto::Sha256Hash;
+
+  private:
+    bool is_set = false;
```

Comment:
Why not making `digest` a `std::optional` instead of having this extra `bool`?

Reply:
It started out that way, but it's very inflexible even now and in the future for extensions/crypto agility.